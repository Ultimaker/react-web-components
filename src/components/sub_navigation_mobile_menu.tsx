import * as React from 'react';

// components
import SlideOutContainer from './slide_out_container';

export interface SubNavigationMobileMenuProps {
    menuLabel: string | JSX.Element | JSX.Element[];
}

export interface SubNavigationMobileMenuState {
    isOpen: boolean;
}

export class SubNavigationMobileMenu extends
    React.Component<SubNavigationMobileMenuProps, SubNavigationMobileMenuState> {
    state = {
        isOpen: false,
    }

    private _toggleShowAccountNav(isOpen: boolean): void {
        this.setState({
            isOpen,
        });
    }

    render(): JSX.Element {
        const { menuLabel, children } = this.props;
        const { isOpen } = this.state;

        return (
            <div className="sub-navigation-mobile-menu">
                <SlideOutContainer
                    isOpen={isOpen}
                    panelArrowWidth="1.2rem"
                    headerText={menuLabel}
                    onHeaderClick={
                        () => this._toggleShowAccountNav(!isOpen)
                    }
                >
                    {children}
                </SlideOutContainer>
            </div>
        );
    }
}

export default SubNavigationMobileMenu;
