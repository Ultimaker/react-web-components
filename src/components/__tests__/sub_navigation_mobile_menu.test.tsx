// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import { SubNavigationMobileMenu, SubNavigationMobileMenuProps } from '../sub_navigation_mobile_menu';
import SlideOutContainer from '../slide_out_container';

describe('The SubNavigationMobileMenu component', () => {
    let props: SubNavigationMobileMenuProps;
    let wrapper;

    beforeEach(() => {
        props = {
            menuLabel: 'Home',
        };
        wrapper = shallow(<SubNavigationMobileMenu {...props} />);
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should show account mobile navigation menu', () => {
        expect(wrapper.find(SlideOutContainer).props().isOpen).toBe(false);
        wrapper.find(SlideOutContainer).props().onHeaderClick();
        expect(wrapper.find(SlideOutContainer).props().isOpen).toBe(true);
    });
});
