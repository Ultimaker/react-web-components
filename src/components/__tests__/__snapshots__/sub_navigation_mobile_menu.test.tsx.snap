// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SubNavigationMobileMenu component should render 1`] = `
<div
  className="sub-navigation-mobile-menu"
>
  <SlideOutContainer
    headerText="Home"
    headerTextAlignment="left"
    isOpen={false}
    onHeaderClick={[Function]}
    panelArrowWidth="1.2rem"
    showHeaderPanelArrow={true}
  />
</div>
`;
